core/example/longrange/CMakeLists.txt

```diff
@@ -0,0 +1,48 @@
+add_executable(Direct directsum_example.cpp)
+target_link_libraries(Direct cabanacore)
+add_executable(Ewald ewaldsum_example.cpp)
+target_link_libraries(Ewald cabanacore)
+
+if(Cabana_ENABLE_Cuda)
+  find_package(CUDA REQUIRED)
+  if(CUDA_FOUND)
+    add_executable(SPME spme_example.cpp)
+    target_link_libraries(SPME cabanacore ${CUDA_CUFFT_LIBRARIES})
+    target_compile_definitions(SPME PUBLIC Cabana_ENABLE_Cuda)
```

Comment:
Once we merge #116, you can get rid of all the `target_compile_definitions()`.

Reply:
That will be a nice improvement. I had to make this change because the previous method (with `add_compile_definitions`, which was slightly cleaner) required CMake 3.12. It's messy for now, but like you say #116 will allow me to clean it up.

Reply:
Done, please go ahead!